velox/functions/prestosql/BinaryFunctions.h

```diff
@@ -436,4 +436,92 @@ struct ToIEEE754Bits32 {
   }
 };
 
+template <bool lpad, typename TOut, typename TIn>
+FOLLY_ALWAYS_INLINE void
+pad(TOut& output, const TIn& input, const int64_t size, const TIn& padding) {
+  static constexpr size_t kMaxPadSize = 1024 * 1024; // 1MB
+  int64_t paddingSize = padding.size();
+
+  VELOX_USER_CHECK(
+      size >= 0 && size <= kMaxPadSize,
+      "Target length must be in the range [0..{})",
+      kMaxPadSize);
+  VELOX_USER_CHECK_GT(paddingSize, 0, "Padding bytes must not be empty");
+
+  int64_t inputSize = input.size();
+  // Input has at most size length, truncate it if necessary
+  // and return it as the result.
+  if (UNLIKELY(inputSize >= size)) {
+    output.resize(size);
+    std::memcpy(output.data(), input.data(), size); // why do we wanna avoid
```

Comment:
What is with the comment at the end of this line?

Reply:
fixed! it was an old todo i didn't remove after resolving